### DISCUSSION TITLE
Are the classes in the stylesheet generated by the CLI listed in the recommended class order?

### DISCUSSION THREAD

**Discussion by airblade** [tailwindlabs/tailwindcss#10343] [Help] [answered]
I know there is a recommended class order when writing classes for an element.

And of course the CLI generates a stylesheet which contains classes among other things.

I was wondering whether the stylesheet's classes are in the recommended class order?

Many thanks!

**Comment by RobinMalfait** [comment, marked as answer]
Hey!

Yep we have an internal order and we have some places where we explicitly sort everything (e.g.: screen variants are ordered based on value, variants have a defined order and stacked variants always combine the variants). In fact, our prettier plugin uses this exact same order we use internally to format and sort your classes in your templates.

A small caveat: sometimes the order is not determenistic but based on the order they first appeared in. However, those don't really matter in real world examples.

E.g.: `bg-red-500` and `bg-green-500`, there is not order or hierarchy attached to these values. If you look at `pl-4`, this will always win from `px-4` which will always win from `p-4`.

Hope this helps!

**Reply by airblade** [reply]
@RobinMalfait Thanks for the info!

So if I understand correctly, both the Prettier plugin and the CLI output classes in the same, recommended order (allowing for the caveat you mentioned)?

**Reply by RobinMalfait** [reply]
Yep! Should be at least 😄

**Reply by airblade** [reply]
Brilliant, thanks again!  I appreciate your time :)
